packages/flutter_tools/lib/src/template.dart

```diff
@@ -328,6 +328,20 @@ class Template {
           context['androidIdentifier'] = _escapeKotlinKeywords(androidIdentifier);
         }
 
+        final bool web = (context['web'] as bool?) ?? false;
+
+        if (web) {
+          // The index.html and manifest.json outputs already have quotes in the template,
+          // so extra quotes are not needed.
+          if (finalDestinationFile.basename == 'index.html' || finalDestinationFile.basename =='manifest.json') {
+            final String? description = context['description'] as String?;
+
+            if (description != null && description.isNotEmpty) {
+              context['description'] = description.replaceAll('"', '');
```

Comment:
Note for myself when doing my next review pass:

Is it safe to modify `context` here? Will this result in bugs if we change the rendering order of the template files? Instead of writing to `context['description']`, maybe we should just make a local copy of it.

Reply:
Then we'd have to make sure that the template renderer gets the local when calling `renderString()`, which might not be as flexible if other things need to get passed down in the future.

I think your idea of delaying the YAML escaping until we have a YAML file to create makes more sense here.
Then it is up to the template renderer to decide when to escape what.

Reply:
Made a local copy of the context when writing to the pubspec.